error in lld.py in line 12

Pusendra opened the pull request:
Traceback (most recent call last):
  File "lld.py", line 12, in <module>
    reload(sys)
NameError: name 'reload' is not defined

miczza committed:
Updated README.md and script name

miczza committed:
Code cleaning and README fix

medtechexchange committed:
- Updating the various regex patterns to use the same file pattern via variable reference in one place.
- Regex pattern now allows for better file saving without as many errors as before on Windows file systems.
- replacement of invalid characters is now a space instead of nothing to help delineate words better
- Fixing typo in readme file.

miczza committed:
Merge pull request #7 from milkshoes/master

Attempt to help fix errors within file paths on Windows with revised regex pattern, additional minor revisions

milkshoes committed:
- Updating regex pattern. this version you suggested via issue tracker seems to works better.

- In the future you could consider something like this below:

invalid_file_chars =  re.compile(r'[^/]')

if os.name == "nt":
   invalid_file_chars = re.compile(r'[^\\/:*?"<>|]')

miczza committed:
Merge pull request #16 from milkshoes/master

Updating regex pattern.

miczza committed:
Added ' to invalid chars.

miczza committed:
Changed encoding to UTF-8

miczza committed:
Added .strip()

miczza committed:
Added utf-8 encoding

miczza committed:
Added utf-8 encoding to chapter name

miczza committed:
Updated .gitignore

duracotton committed:
Add files via upload

Subtitle Support, Release Date, Chapter & File numbering

duracotton committed:
Subtitle, Release-Date, File numbering, Umlaut

Added Subtitle-Downloads, Release-Date and Chapter/Video-Numbering, German Umlaut

duracotton committed:
subtitle support, releasedate, file numbering

1. The course release date is now added to the course folder's name (because Lynda often updated the courses and you didn't know of what date your downloaded version was). In the future, we should differentiate between the release-date element and the update-date element (which's name I don't know yet) and use the newer one of both for the folder name.

2. I added subtitle support. That was a bit of a mess, but I tried my best, seems to be working. .srt files have to be the same name of the .mp4 files so a player like MPC HC will load it automatically.

3. I took over the chapter/video numbering from https://github.com/mclmza/linkedin-learning-downloader/issues/3#issuecomment-349475909, so thanks for that!

4. Merged pull requests https://github.com/mclmza/linkedin-learning-downloader/pull/36 and https://github.com/mclmza/linkedin-learning-downloader/pull/33

5. Corrected bug: re.sub(invalid_file_chars, " ", string) is leading to an IO error when a video has an unallowed char at the end, e.g. out\What Is the Google Cloud Platform \ but Windows will remove space at the end of an folder, so the next download attempt will fail because there isn't such a path anymore. Strip() does the thing. Just saw that https://github.com/mclmza/linkedin-learning-downloader/pull/33 also had that idea.

6. Replaced German Umlaut with their non-crictial equivalents

duracotton committed:
exercise download, check for full access, etc

1. Added download for exercise files
2. Checking now for full course access
3. Now using configurable download path
4. Some code clean-up

duracotton committed:
Update README.md

duracotton committed:
Description download, bugfixes

duracotton committed:
Catching 404/elements error

Now catching the 'elements' error when trying to loop through an empty list because the request gave an 404 response caused by mostly user error when inserting a wrong course slug in the config file.

duracotton committed:
Bugfix

duracotton committed:
Bugfixes again

duracotton committed:
double-digit zfill of file numbers, HTTP 429

duracotton committed:
Parsing bookmarks, disabling downloaded courses

1. Easier adding of courses to the downloader. Just add them via LinkedIn (app or browser) to your bookmarks. Tool will now parse your LinkedIn bookmarks (https://www.linkedin.com/learning/me/saved).
The previously alternative of adding course slugs manually by pasting them into the config file is still there. The app will do a check for duplicates before automatically adding a course-slug to the list.
2. After successfully downloading a course, the course will be disabled in the config file (Download list) by putting a "#" in front of the course. If you want to check the files for integrity again, just the "#" and the next time when running the app it will check the enabled courses (if already).

duracotton committed:
Bugfix

duracotton committed:
Update README.md

duracotton committed:
Update README.md

duracotton committed:
Workaround: LinkedIn providing only 10 bookmarks

Fixed Bug: Built a workaround for getting all bookmarks. LinkedIn is only providing 10 bookmarks at a time, so API has to be triggered 
(bookmarks/10) times

duracotton committed:
Update README.md

duracotton committed:
Update README.md

duracotton committed:
Update README.md

duracotton committed:
Update README.md

dpkkdn committed:
check presence of course with older folder name; add functions to rename old folders and files, use extractName to get alphabetic names. Backwards compatible with previous folder structure and naming convention. Allows to complete previous downloads (with old naming of folders and files) and change the naming of folder and files. https://github.com/mclmza/linkedin-learning-downloader/issues/40.

dpkkdn committed:
check presence of course with older folder name; add functions to rename old folders and files, use extractName to get alphabetic names. Backwards compatible with previous folder structure and naming convention. Allows to complete previous downloads (with old naming of folders and files) and change the naming of folder and files

dpkkdn committed:
Merge branch 'backwards-compatible-naming-resume' of https://github.com/dpkkdn/linkedin-learning-downloader into backwards-compatible-naming-resume

miczza committed:
Merge pull request #39 from duracotton/master

Various additions: subtitle & exercise downloads, file numbering, bookmark parsing etc

miczza committed:
Merge pull request #44 from dpkkdn/backwards-compatible-naming-resume

Backwards compatible naming resume

miczza committed:
Setted 'out' as default folder for download

miczza committed:
Some comment cleaning

miczza committed:
Forgot an assignment

miczza committed:
Code refactored

miczza committed:
Updated config.py

miczza committed:
Fix no named chapter bug

miczza committed:
Fixed #76

miczza committed:
Exercise files issue (#75) fixed

miczza committed:
Indentation  fix (#75)

miczza committed:
An other indentation fix (#75)

circuitgeeks commented:
File "llcd.py", line 55
    print '[*] Obtained new session: %s' % session
                                       ^
SyntaxError: Missing parentheses in call to 'print'. Did you mean print(print '[*] Obtained new session: %s' % session)?

I get this error Everytime

niels-numbers committed:
Skip existing videos

SuperWizard7 commented:
I'm getting an error:
**Traceback (most recent call last):
File "lld.py", line 188, in <module>,
main()
File "lld.py", line 183, in main,
lld.get_logged_session()
File "lld.py", line 105, in get_logged_session 
csrf = login_page.find(id='loginCsrfParam-login')['value']
TypeError: 'NoneType' object is not subscriptable**

Please help.

CuongGIK commented:
Please help me, this errors: 
Traceback (most recent call last):
  File "lld.py", line 187, in <module>
    main()
  File "lld.py", line 182, in main
    lld.get_logged_session()
  File "lld.py", line 104, in get_logged_session
    csrf = login_page.find(id='loginCsrfParam-login')['value']
TypeError: 'NoneType' object has no attribute '__getitem__'

zeroplus1994 commented:
> Traceback (most recent call last):
> File "lld.py", line 12, in 
> reload(sys)
> NameError: name 'reload' is not defined

I think it works only on up to Python 2.7 ! No support for Python 3.

sonnb committed:
Update lld.py

A minor changes to make it works with the current LinkedIn form.

sonnb committed:
Update lld.py

Added 1 second sleep between requests to server to prevent 429 response (Too many requests).

ha100 committed:
Merge pull request #1 from iwasherefirst2/master

Skip existing videos

ha100 committed:
Merge pull request #4 from sonnb/master

This should fix the download issue

fastmarketo commented:
Line 50 is throwing an error:

![image](https://user-images.githubusercontent.com/6827222/94379113-f0428b80-014b-11eb-8837-ec86742fb94d.png)

Jenishts commented:
> Line 50 is throwing an error:
> 
> ![image](https://user-images.githubusercontent.com/6827222/94379113-f0428b80-014b-11eb-8837-ec86742fb94d.png)

**You are geting this error becuase you are using python3 but this code it designed to run on python2. I think you have installed python2 and 3, so you will need to mentioned all the command as python2 and start if from the beginning**
